### PR TITLE
Shutdown / Restart refactor

### PR DESCRIPTION
**Pull request by JaiZed**
- centralize and simplify code
- eliminate code duplication
- hide unnecessary details from rest of code by abstraction
- standardize exit codes
- shorten shutdown / restart time
- improve console messaging

### CODE REVIEW

**Comment by morpheus65535** [comment]
This one must be tested under Linux and macOS once merge conflict is fixed.

**Comment by morpheus65535** [comment]
Merge conflict? Possible to get it fixed so I can test this on macos and Linux?

File: bazarr/init.py [hunk 1]
```diff
@@ -99,19 +96,20 @@ def is_virtualenv():
...
-                    restart_file = io.open(os.path.join(args.config_dir, "bazarr.restart"), "w", encoding='UTF-8')
-                except Exception as e:
-                    logging.error(f'BAZARR Cannot create restart file: {repr(e)}')
-                else:
-                    logging.info('Bazarr is being restarted...')
-                    restart_file.write(str(''))
-                    restart_file.close()
-                    os._exit(0)
+                restart_bazarr()
+                # try:
```
Comment (morpheus65535) [inline bazarr/init.py:104]: This commented block is expected?

File: bazarr.py [hunk 2]
```diff
@@ -49,55 +52,77 @@ def get_python_path():
...
+def interrupt_handler(signum, frame):
+    # catch and ignore keyboard interrupt Ctrl-C
+    # the child process Server object will catch SIGINT and perform an orderly shutdown
+    global interrupted
+    if not interrupted:
+        # ignore user hammering Ctrl-C; we heard you the first time!
+        interrupted = True
+        print('Handling keyboard interrupt...')
+    else:
+        print(f"Stop doing that! I heard you the first time!")
```
Comment (morpheus65535) [inline bazarr.py:116]: Nice one ;-)

File: bazarr/init.py (hunk 1 above)
Comment (JaiZed) [reply bazarr/init.py:104]: No.
It was a reminder of what the code was doing before the refactoring.
I will remove it.

File: bazarr.py (hunk 2 above)
Comment (JaiZed) [reply bazarr.py:116]: It actually does help if the user does that.
Otherwise, it would “restart” the shutdown process.

**Pull request merged**